File: tps/entrada e saida/tp01/TP01Q01 - Palíndromo em Java/Palindromo.java
class Palindromo {

   public static boolean isFim(String s){
      return (s.length() == 3 && s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M');
   }

   public static String invertida (String invert, String s){
     for (int i = (s.length()-1); i >= 0; i--) {
        invert = invert + s.charAt(i);
     }
     
     return invert;
   }

   public static boolean isPalindromo (String s){
      String invert = new String();
      boolean isP = true;
      
      invert = invertida("", s);
 
      for (int i = 0; i < s.length(); i++) {
        if(invert.charAt(i) == s.charAt(i) && isP == true)
        {
          isP = true;
        } else {
          isP = false;
        }
      }

      return isP;
   }

   public static void main (String[] args){
      String[] entrada = new String[1000];
      int numEntrada = 0;

      //Leitura da entrada padrao
      do {
         entrada[numEntrada] = MyIO.readLine();
      } while (isFim(entrada[numEntrada++]) == false);
      numEntrada--;   //Desconsiderar ultima linha contendo a palavra FIM

      //Para cada linha de entrada, gerando uma de saida boleana se e' ou nao palindromo
      for(int i = 0; i < numEntrada; i++){
        if(isPalindromo(entrada[i])){
          MyIO.println("SIM");
        } else {
          MyIO.println("NAO");
        }
      }
   }
}

File: bin/tps/entrada e saida/tp01/TP01Q02 - Palíndromo em C/Palindromo.c
#include<stdio.h>
#include<stdbool.h>
#include<string.h>

bool isPalindromo (char s[], int i)
{
    char invert[1000];
    int size = i;
    int palin = 0;
    bool is = false;
    i--;

    for (int j = 0; j < size; j++)
    {
        invert[j] = s[i];
        i--;
    }
    
    for (int a = 0; a <= size; a++)
    {
        if (s[a] == invert[a])
        {
            palin++;
        }
    }

    if (palin == size)
    {
        is = true;
    }
    return(is);   
    
}

bool isFim(char s[], int i)
{
    return (i == 3 && s[0] == 'F' && s[1] == 'I' && s[2] == 'M');
}

int main( )
{
    int size = 0;
    char s[1000];
    int i = 0;

    do
    {
        fgets(s, 1000, stdin);
        size = strlen(s)-1; 
        
        if (!isFim(s, size))
        {
            if (isPalindromo(s, size))
            {
                printf("SIM\n");
            } 
            else
            {
                printf("NAO\n");
            } 
        }

    }while (!isFim(s, size));
 return 0;
}